catt/controllers.py

```diff
@@ -61,24 +61,24 @@ def all_info(self):
 ]
 
 
-def get_chromecasts() -> List[pychromecast.Chromecast]:
-    devices = pychromecast.get_chromecasts()
+def get_chromecasts_and_ips(names: Optional[List[str]] = None) -> List[Tuple[pychromecast.Chromecast, str]]:
+    if names:
+        services, browser = pychromecast.discovery.discover_listed_chromecasts(friendly_names=names)
+    else:
+        services, browser = pychromecast.discovery.discover_chromecasts()
+    pychromecast.stop_discovery(browser)
+    return [(pychromecast.get_chromecast_from_service(s, browser.zc), s[4]) for s in services]
+
+
+def get_chromecasts(names: Optional[List[str]] = None) -> List[pychromecast.Chromecast]:
```

Comment:
These functions need a line or two of docstring, probably. Most are fairly straightforward in what they do, but there are some considerations (e.g. speed of discovery, whether they connect to the Chromecast or not, what a CCInfo object is) that isn't currently clear.